### PR TITLE
Update workflows

### PR DESCRIPTION
**Pull request by bonddim**
Main improvements:
- Manage GitHub actions updates with dependabot
- Update all actions to latest versions
- Merge GoReleaser workflows into one
- Add groups to release changelog (goreleaser config)
- Add workflow to cleanup PR caches
- Use go with version from `go.mod` file

### CODE REVIEW

**Comment by yxxhero** [comment]
@bonddim please fix DCO issue.

**Comment by yxxhero** [comment]
@bonddim awesome work. thanks so much.

**Review by yxxhero — APPROVED** [review]
LGTM

**Comment by yxxhero** [comment]
@mumoshu WDYT?

File: .goreleaser.yml [hunk 1]
```diff
@@ -23,5 +23,19 @@ builds:
       - amd64
       - arm64
       - "386"
+
 changelog:
-  use: github-native
+  use: github
```
Comment (yxxhero) [inline .goreleaser.yml:28]: why change this?

Comment (bonddim) [reply .goreleaser.yml:28]: This supports groups in changelog.

**Pull request merged**